pkg/alertmanager/amcfg.go

```diff
@@ -128,6 +128,56 @@ func validateConfigInputs(am *monitoringv1.Alertmanager) error {
 	return nil
 }
 
+func (cg *configGenerator) generateGlobalConfig(
+	ctx context.Context,
+	amConfig *monitoringv1alpha1.AlertmanagerConfig,
+) (*alertmanagerConfig, error) {
+	globalAlertmanagerConfig := &alertmanagerConfig{
+		Receivers: []*receiver{
+			{
+				Name: "null",
+			},
+		},
+		Route: &route{
+			Receiver: "null",
+		},
+	}
+	crKey := types.NamespacedName{
+		Namespace: amConfig.Namespace,
+		Name:      amConfig.Name,
+	}
+	// Add inhibitRules to baseConfig.InhibitRules without enforce namespace
+	for _, inhibitRule := range amConfig.Spec.InhibitRules {
+		globalAlertmanagerConfig.InhibitRules = append(globalAlertmanagerConfig.InhibitRules, cg.convertInhibitRule(&inhibitRule))
+	}
+
+	// Add routes to baseConfig.Route.Routes without enforce namespace
+	globalAlertmanagerConfig.Route.Routes = []*route{
+		cg.convertRoute(amConfig.Spec.Route, crKey),
+	}
```

Comment:
We should update the root entry
```suggestion
	globalAlertmanagerConfig.Route = cg.convertRoute(amConfig.Spec.Route, crKey)
```